src/graph/UpdateEdgeExecutor.h

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef GRAPH_UPDATEEDGEEXECUTOR_H_
+#define GRAPH_UPDATEEDGEEXECUTOR_H_
+
+#include "base/Base.h"
+#include "filter/Expressions.h"
+#include "graph/Executor.h"
+#include "meta/SchemaManager.h"
+#include "storage/client/StorageClient.h"
+
+namespace nebula {
+
+namespace storage {
+
+namespace cpp2 {
+class UpdateResponse;
+}   // namespace cpp2
+}   // namespace storage
+
+namespace graph {
+
+class UpdateEdgeExecutor final : public Executor {
+public:
+    UpdateEdgeExecutor(Sentence *sentence, ExecutionContext *ectx);
+
+    const char* name() const override {
+        return "UpdateEdgeExecutor";
+    }
+
+    Status MUST_USE_RESULT prepare() override;
+
+    void execute() override;
+
+    void setupResponse(cpp2::ExecutionResponse &resp) override;
+
+private:
+    // To do some preparing works on the clauses
+    Status prepareSet();
+
+    Status prepareWhere();
+
+    Status prepareYield();
+
+    Status prepareNeededProps();
+
+    StatusOr<std::vector<storage::cpp2::PropDef>> getReturnProps();
+
+    void insertReverselyEdge();
+
+    // All required data have arrived, finish the execution.
+    void finishExecution(storage::cpp2::UpdateResponse &&rpcResp);
+
+    /**
+     * A container to hold the mapping from vertex id to its properties, used for lookups
+     * during the final evaluation process.
+     */
+    class VertexHolder final {
```

Comment:
Why we need VertexHolder?  There is only one vertex or edge

Reply:
Wrong code, it has been removed!